components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationMetadataService.java

```diff
@@ -0,0 +1,325 @@
+                        .options(idTokenAlgorithmDTO.getSupportedIdTokenEncryptionAlgorithms()));
+        oidcMetaData.idTokenEncryptionMethod(
+                new MetadataProperty()
+                        .defaultValue(idTokenAlgorithmDTO.getDefaultIdTokenEncryptionMethod())
+                        .options(idTokenAlgorithmDTO.getSupportedIdTokenEncryptionMethods()));
+
+        oidcMetaData.setScopeValidators(
+                new MetadataProperty()
+                        .defaultValue(null)
+                        .options(Arrays.asList(oAuthAdminService.getAllowedScopeValidators())));
+
+        oidcMetaData.accessTokenType(
+                new MetadataProperty()
+                        .defaultValue(oAuthAdminService.getDefaultTokenType())
+                        .options(oAuthAdminService.getSupportedTokenTypes()));
+
+        return oidcMetaData;
+    }
+
+    /**
+     * Pull WS Trust metadata from STSAdminServiceInterface and return.
+     *
+     * @return Populated WSTrustMetadata object.
+     */
+    public WSTrustMetaData getWSTrustMetadata() {
+
+        WSTrustMetaData wsTrustMetaData = new WSTrustMetaData();
+        try {
+            wsTrustMetaData.setCertificateAlias(new MetadataProperty()
+                    .defaultValue(null)
+                    .options(Arrays.asList(
+                            ApplicationManagementServiceHolder.getStsAdminService().getCertAliasOfPrimaryKeyStore())));
+        } catch (SecurityConfigException e) {
+            throw handleException(e, ERROR_WHILE_RETRIEVING_SAML_METADATA);
+        }
+        return wsTrustMetaData;
+    }
+
+    /**
+     * Pull property metadata of the custom inbound protocol that matches to the protocol name.
+     *
+     * @param inboundProtocolId Base64URL encoded protocol name.
+     * @return Populated CustomInboundProtocolMetaData object.
+     */
+    public CustomInboundProtocolMetaData getCustomProtocolMetadata(String inboundProtocolId) {
+
+        String protocolName = base64URLDecode(inboundProtocolId);
+        Map<String, AbstractInboundAuthenticatorConfig> allCustomAuthenticators =
+                ApplicationManagementServiceHolder.getApplicationManagementService().getAllInboundAuthenticatorConfig();
+
+        // Loop through all custom inbound protocols and match the name.
+        for (Map.Entry<String, AbstractInboundAuthenticatorConfig> entry : allCustomAuthenticators
+                .entrySet()) {
+            if (entry.getValue().getName().equals(protocolName)) {
+                return new CustomInboundProtocolMetaData()
+                        .displayName(entry.getValue().getFriendlyName())
+                        .properties(getCustomInboundProtocolProperties(entry.getValue().getConfigurationProperties()));
+            }
+        }
+
+        // Throw 404 error if the protocol not found
+        throw buildApiError(ApplicationManagementConstants.ErrorMessage.ERROR_INBOUND_PROTOCOL_NOT_FOUND);
```

Comment:
What should be the correct error code for this? 400 range or 500 range?

Reply:
Since this is a client error, the error code will be 600xx. All error codes defined in the constant class will be refined by @mefarazath after finalizing some issues.